File: src/main/java/com/maxciv/infer/plugin/toolwindow/ToolWindowFactory.java
package com.maxciv.infer.plugin.toolwindow;

import com.intellij.openapi.project.Project;
import com.intellij.openapi.wm.ToolWindow;
import com.intellij.openapi.wm.ToolWindowType;
import com.intellij.ui.content.Content;
import org.jetbrains.annotations.NotNull;

/**
 * @author maxim.oleynik
 * @since 28.11.2018
 */
public class ToolWindowFactory implements com.intellij.openapi.wm.ToolWindowFactory {

    @Override
    public void createToolWindowContent(@NotNull Project project, @NotNull ToolWindow toolWindow) {
        Content results = toolWindow.getContentManager().getFactory().createContent(
                new ResultsTab(project),
                "Results",
                false);
        toolWindow.getContentManager().addContent(results);

        Content settings = toolWindow.getContentManager().getFactory().createContent(
                new SettingsTab(project),
                "Settings",
                false);
        toolWindow.getContentManager().addContent(settings);

        toolWindow.setType(ToolWindowType.DOCKED, null);
    }
}

File: src/main/kotlin/com/maxciv/infer/plugin/ui/toolwindow/ToolWindowFactory.kt
package com.maxciv.infer.plugin.ui.toolwindow

import com.intellij.openapi.project.Project
import com.intellij.openapi.wm.ToolWindow
import com.intellij.openapi.wm.ToolWindowType

/**
 * @author maxim.oleynik
 * @since 28.11.2018
 */
class ToolWindowFactory : com.intellij.openapi.wm.ToolWindowFactory {

    override fun createToolWindowContent(project: Project, toolWindow: ToolWindow) {
        val results = toolWindow.contentManager.factory.createContent(
                ResultsTab(project),
                "Results",
                false)
        toolWindow.contentManager.addContent(results)

        val settings = toolWindow.contentManager.factory.createContent(
                SettingsTab(project),
                "Settings",
                false)
        toolWindow.contentManager.addContent(settings)

        toolWindow.setType(ToolWindowType.DOCKED, null)
    }
}
